bika/lims/content/analysis.py

```diff
@@ -32,15 +34,31 @@ def getSample(self):
             return sample
 
     @security.public
-    def getSiblings(self):
+    def getSiblings(self, retracted=False):
+        """
+        Returns the list of analyses of the Analysis Request to which this
+        analysis belongs to, but with the current analysis excluded.
+        :param retracted: If false, retracted/rejected siblings are dismissed
+        :type retracted: bool
+        :return: list of siblings for this analysis
+        :rtype: list of IAnalysis
+        """
         """Returns the list of analyses of the Analysis Request to which this
         analysis belongs to, but with the current analysis excluded
         """
-        siblings = []
         request = self.getRequest()
-        if request:
-            ans = request.getAnalyses(full_objects=True)
-            siblings = [an for an in ans if an.UID() != self.UID()]
+        if not request:
+            return []
+
+        siblings = []
+        retracted_states = [STATE_RETRACTED, STATE_REJECTED]
+        ans = request.getAnalyses(full_objects=True)
+        for sibling in ans:
+            if sibling.UID() == self.UID():
+                continue
+            if retracted == False and in_state(sibling, retracted_states):
```

Comment:
better use `if retracted is False and in_state(sibling, retracted_states)` (compare bool with `is`)